bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/protocol/LuxomCommunication.java

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.luxom.internal.protocol;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.InetAddress;
+import java.net.Socket;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.luxom.internal.handler.LuxomBridgeHandler;
+import org.openhab.binding.luxom.internal.handler.LuxomConnectionException;
+import org.openhab.binding.luxom.internal.handler.config.LuxomBridgeConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LuxomCommunication} class is able to do the following tasks with Luxom IP
+ * systems:
+ * <ul>
+ * <li>Start and stop TCP socket connection with Luxom IP-interface.
+ * <li>Read all setup and status information from the Luxom Controller.
+ * <li>Execute Luxom commands.
+ * <li>Listen to events from Luxom.
+ * </ul>
+ *
+ * @author Kris Jespers - Initial contribution
+ */
+@NonNullByDefault
+public class LuxomCommunication {
+
+    private final Logger logger = LoggerFactory.getLogger(LuxomCommunication.class);
+
+    private final LuxomBridgeHandler bridgeHandler;
+
+    private @Nullable Socket luxomSocket;
+    private @Nullable PrintWriter luxomOut;
+    private @Nullable BufferedReader luxomIn;
+
+    private volatile boolean listenerStopped;
+    private volatile boolean stillListeningToEvents;
+
+    public LuxomCommunication(LuxomBridgeHandler luxomBridgeHandler) {
+        super();
+        this.bridgeHandler = luxomBridgeHandler;
+    }
+
+    public synchronized void startCommunication() throws LuxomConnectionException {
+        try {
+            waitForEventListenerThreadToStop();
+
+            initializeSocket();
+
+            // Start Luxom event listener. This listener will act on all messages coming from
+            // IP-interface.
+            (new Thread(this::runLuxomEvents)).start();
```

Comment:
Can you also give the thread a name?
That helps with trouble shooting threading issues.
The convention is to use `OH-binding-<bridge/thing-uid>-<purpose>`